### DISCUSSION TITLE
Image Normalization

### DISCUSSION THREAD

**Discussion by saeidn95** [vladmandic/human#445] [Q&A]
I think the constant.tf05 in (https://github.com/vladmandic/human/blob/62396317f5c31b2ec5bc7de2e9eb53b8978b9f28/src/face/blazeface.ts#L60) should be constant.tf1.

I am basing this on the google tensorflow/tfjs-models/blazeface/src/face.ts ( https://github.com/tensorflow/tfjs-models/blob/c5fcabec4cc0335bdfabeb14dc2adfd6ef0cde8c/blazeface/src/face.ts#L239)

But I could be wrong!

**Comment by vladmandic** [comment]
```js
  t.resized = tf.image.resizeBilinear(inputImage, [inputSize, inputSize]);
  t.div = tf.div(t.resized, constants.tf127);
  t.normalized = tf.sub(t.div, constants.tf05);
```

normalized = image / 127 - 0.5

```js
      const normalizedImage = tf.mul(tf.sub(tf.div(resizedImage, 255), 0.5), 2);
```

normalized = (image / 255 - 0.5) * 2

i think you might be right - care to create a pr since you found it?

**Reply by saeidn95** [reply]
On second though you will need to double check this. The blaze model used in (https://github.com/tensorflow/tfjs-models/blob/c5fcabec4cc0335bdfabeb14dc2adfd6ef0cde8c/blazeface/src/index.ts) is a 128x128 input model. where as the model blaze model used in human is 256x256. It is unlikely, but, they might have a different normalization requirement.

**Comment by vladmandic** [comment]
reviewed, you were right originally, it should be `tf1`, new code will be uploaded shortly.
